e2e/tests/dump-restore-test.rs

```diff
@@ -0,0 +1,304 @@
+        exit
+    );
+    println!("database snapshot at: {}", path.display());
+    fs::read_to_string(path).unwrap()
+}
+
+fn dump_db(container_id: &str, db: &str, username: &str) -> PathBuf {
+    let exit = Command::new("docker")
+        .arg("exec")
+        .arg(container_id)
+        .arg("pg_dump")
+        .arg("-U")
+        .arg(username)
+        .arg("-d")
+        .arg(db)
+        .arg("-v")
+        .arg("-F")
+        .arg("p")
+        .arg("-T")
+        .arg("_prom_catalog.\"default\"") // todo: undecided atm
+        .arg("-T")
+        .arg("_prom_catalog.ids_epoch") // todo: Mat to look at
+        .arg("-T")
+        .arg("_prom_catalog.remote_commands") // todo: set seq to start at 1000, only dump records >= 1000
+        .arg("-T")
+        .arg("_ps_catalog.migration") // todo: should NOT be a config table. must restore at the same version at which we dumped
+        .arg("-T")
+        .arg("_ps_catalog.promscale_instance_information") // todo: should NOT be a config table
+        .arg("-T")
+        .arg("_ps_trace.tag_key") // todo: only dump records >= 1000
+        .arg("-T")
+        .arg("public.prom_installation_info") // todo: ???
+        .arg("-N")
+        .arg("_timescaledb_internal")
+        .arg("-N")
+        .arg("_timescaledb_cache")
+        .arg("-N")
+        .arg("_timescaledb_catalog")
+        .arg("-N")
+        .arg("_timescaledb_config")
+        .arg("-N")
+        .arg("_timescaledb_internal")
+        .arg("-N")
+        .arg("timescaledb_experimental")
+        .arg("-N")
+        .arg("timescaledb_information")
+        .arg("-f")
+        .arg("/tmp/dump.sql")
+        .spawn()
+        .unwrap()
+        .wait()
+        .unwrap();
+    assert!(
+        exit.success(),
+        "dumping the database {} via {} user failed: {}",
+        db,
+        username,
+        exit
+    );
+    let path = env::temp_dir().join("dump.sql");
+    if path.exists() {
+        fs::remove_file(path.clone()).expect("failed to remove existing dump file");
+    }
+    let exit = Command::new("docker")
+        .arg("cp")
+        .arg(format!("{}:/tmp/dump.sql", container_id))
+        .arg(path.to_str().unwrap())
+        .spawn()
+        .unwrap()
+        .wait()
+        .unwrap();
+    assert!(
+        exit.success(),
+        "copying the dump out of the container failed: {}",
+        exit
+    );
+    path
+}
+
+fn restore_db(container_id: &str, db: &str, username: &str, path: PathBuf) {
+    let exit = Command::new("docker")
+        .arg("cp")
+        .arg(path.to_str().unwrap())
+        .arg(format!("{}:/tmp/dump.sql", container_id))
+        .spawn()
+        .unwrap()
+        .wait()
+        .unwrap();
+    assert!(
+        exit.success(),
+        "copying the dump into the container failed: {}",
+        exit
+    );
+    let exit = Command::new("docker")
+        .arg("exec")
+        .arg(container_id)
+        .arg("psql")
+        .arg("-U")
+        .arg(username)
```

Comment:
Nit: not sure if rustfmt allows this but would be nice to group args like this.
```suggestion
        .arg("-U").arg(username)
```

Reply:
I had the same thought and tried it, but it complained :(

Reply:
oh, but  I can do this: `.args(["-v", "ON_ERROR_STOP=1"])` :fire: